src/tools/rbd_wnbd/rbd_wnbd.cc

```diff
@@ -902,6 +902,42 @@ class RBDService : public ServiceBase {
     }
 };
 
+class WNBDWatchCtx : public librbd::UpdateWatchCtx
+{
+private:
+  librados::IoCtx &io_ctx;
+  WnbdHandler* handler;
+  librbd::Image &image;
+  uint64_t size;
+public:
+  WNBDWatchCtx(librados::IoCtx& io_ctx, WnbdHandler* handler,
+               librbd::Image& image, uint64_t size)
+    : io_ctx(io_ctx)
+    , handler(handler)
+    , image(image)
+    , size(size)
+  { }
+
+  ~WNBDWatchCtx() override {}
+
+  void handle_notify() override
+  {
+    librbd::image_info_t info;
+    if (image.stat(info, sizeof(info)) == 0) {
+      uint64_t new_size = info.size;
+      int ret;
+
+      if (new_size != size) {
+        ret = handler->resize(new_size);
+	      
+        if (!ret)
+          size = new_size;
```

Comment:
Nit: `ret` variable is redundant:
```suggestion
        if (!handler->resize(new_size))
          size = new_size;
```